ui/admin/app/components/model-providers/ModelProviderConfigure.tsx

```diff
@@ -0,0 +1,107 @@
+import { BoxesIcon, SettingsIcon } from "lucide-react";
+import { useState } from "react";
+import useSWR from "swr";
+
+import { ModelProvider, ModelProviderConfig } from "~/lib/model/modelProviders";
+import { ModelProviderApiService } from "~/lib/service/api/modelProviderApiService";
+
+import { ModelProviderForm } from "~/components/model-providers/ModelProviderForm";
+import { LoadingSpinner } from "~/components/ui/LoadingSpinner";
+import { Button } from "~/components/ui/button";
+import {
+    Dialog,
+    DialogContent,
+    DialogDescription,
+    DialogHeader,
+    DialogTitle,
+    DialogTrigger,
+} from "~/components/ui/dialog";
+
+type ModelProviderConfigureProps = {
+    modelProvider: ModelProvider;
+};
+
+export function ModelProviderConfigure({
+    modelProvider,
+}: ModelProviderConfigureProps) {
+    const [dialogIsOpen, setDialogIsOpen] = useState(false);
+    const handleOpenChange = (open: boolean) => {
+        if (!open) {
+            setDialogIsOpen(false);
+        }
+    };
+
+    return (
+        <Dialog open={dialogIsOpen} onOpenChange={handleOpenChange}>
+            <DialogTrigger asChild>
+                <Button
+                    size="icon"
+                    variant="ghost"
+                    className="mt-0"
+                    onClick={() => setDialogIsOpen(true)}
+                >
+                    <SettingsIcon />
+                </Button>
+            </DialogTrigger>
```

Comment:
nit: `DialogTrigger` handles opening/closing the dialog automatically
```suggestion
    const [dialogIsOpen, setDialogIsOpen] = useState(false);
    return (
        <Dialog open={dialogIsOpen} onOpenChange={setDialogIsOpen}>
            <DialogTrigger asChild>
                <Button
                    size="icon"
                    variant="ghost"
                    className="mt-0"
                >
                    <SettingsIcon />
                </Button>
            </DialogTrigger>
```